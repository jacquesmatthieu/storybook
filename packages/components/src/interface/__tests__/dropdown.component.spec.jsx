import React from 'react';
import { shallow } from 'enzyme';

import Dropdown from '../dropdown.component';

describe('Dropdown', () => {
  let props;

  beforeEach(() => {
    props = {};
  });

  const getWrapper = () => shallow(<Dropdown {...props}>{({ open }) => `opened: ${open}`}</Dropdown>);

  it('should render properly', () => {
    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle close', () => {
    // Given
    const wrapper = getWrapper();
    wrapper.setState({ open: true });

    // When
    wrapper.find('dropdowncomponent__DropdownClose').simulate('click');

    // Then
    expect(wrapper).toMatchSnapshot();
  });
});
