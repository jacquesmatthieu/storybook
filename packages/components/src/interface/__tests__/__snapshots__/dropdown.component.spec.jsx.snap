// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should handle close 1`] = `
<div>
  opened: false
</div>
`;

exports[`Dropdown should render properly 1`] = `
<div>
  opened: false
</div>
`;
